core/commands/dht.go

```diff
@@ -39,13 +45,191 @@ var DhtCmd = &cmds.Command{
 		"get":       getValueDhtCmd,
 		"put":       putValueDhtCmd,
 		"provide":   provideRefDhtCmd,
+		"stat":      statDhtCmd,
 	},
 }
 
 const (
 	dhtVerboseOptionName = "verbose"
 )
 
+type dhtPeerInfo struct {
+	ID            string
+	Connected     bool
+	AgentVersion  string
+	LastUsefulAt  string
+	LastQueriedAt string
+}
+
+type dhtStat struct {
+	Name    string
+	Buckets []dhtBucket
+}
+
+type dhtBucket struct {
+	LastRefresh string
+	Peers       []dhtPeerInfo
+}
+
+var statDhtCmd = &cmds.Command{
```

Comment:
given the level of information here, is it more sensible to have this live as `ipfs diag dht`?

Reply:
Possibly? I'm trying to mirror `ipfs bitswap stat` but I agree that lumping this all under the `diag` command probably makes more sense (and really, that's where the `ipfs bitswap` commands should live as well.

Reply:
Actually, I'm not sure. `diag` is mostly for true diagnostics (running commands, system information, possibly pprof dumps in the future, etc.).